src/Raven.Server/Documents/Patch/ScriptRunner.cs

```diff
@@ -1819,16 +1818,17 @@ private JsValue LoadDocumentInternal(string id)
             private JsValue[] _args = Array.Empty<JsValue>();
             private readonly JintPreventResolvingTasksReferenceResolver _refResolver = new JintPreventResolvingTasksReferenceResolver();
 
-            public ScriptRunnerResult Run(JsonOperationContext jsonCtx, DocumentsOperationContext docCtx, string method, object[] args, QueryTimingsScope scope = null)
+            public ScriptRunnerResult Run(JsonOperationContext jsonCtx, DocumentsOperationContext docCtx, string method, object[] args, QueryTimingsScope scope = null, CancellationToken token = default)
             {
-                return Run(jsonCtx, docCtx, method, null, args, scope);
+                return Run(jsonCtx, docCtx, method, null, args, scope, token);
             }
 
-            public ScriptRunnerResult Run(JsonOperationContext jsonCtx, DocumentsOperationContext docCtx, string method, string documentId, object[] args, QueryTimingsScope scope = null)
+            public ScriptRunnerResult Run(JsonOperationContext jsonCtx, DocumentsOperationContext docCtx, string method, string documentId, object[] args, QueryTimingsScope scope = null, CancellationToken token = default)
             {
                 _docsCtx = docCtx;
                 _jsonCtx = jsonCtx ?? ThrowArgumentNull();
                 _scope = scope;
+                _token = token;
```

Comment:
We need to clear that in the `finally`, because instances are reused

Reply:
done